VirtoCommerce.CatalogModule.Data/Services/CategoryServiceImpl.cs

```diff
@@ -111,6 +111,11 @@ public virtual void Delete(string[] categoryIds)
 
                 _eventPublisher.Publish(new CategoryChangedEvent(changedEntries));
             }
+
+            foreach (var category in categories)
```

Comment:
This logic doesn't remove the seo info  for nested categories need to expand categories tree and call DeleteSeoForObject for these objects.